crates/topos-tce/src/app_context/network.rs

```diff
@@ -21,124 +21,122 @@ impl AppContext {
-                                self.delivery_latency.entry(cert.id)
-                            {
-                                entry.insert(CERTIFICATE_DELIVERY_LATENCY.start_timer());
-                            }
-
-                            spawn(async move {
-                                info!("Send certificate {} to be broadcast", cert.id);
-                                if channel
-                                    .send(DoubleEchoCommand::Broadcast {
-                                        cert,
-                                        need_gossip: false,
-                                    })
-                                    .await
-                                    .is_err()
-                                {
-                                    error!(
-                                        "Unable to send broadcast_new_certificate command, \
-                                         Receiver was dropped"
-                                    );
-                                }
-                            });
-                        }
-                        Err(e) => {
-                            error!("Error converting received certificate {e}");
-                        }
-                    },
-                    double_echo_request::Request::Echo(Echo {
-                        certificate_id: Some(certificate_id),
-                        signature: Some(signature),
-                        validator_id: Some(validator_id),
-                    }) => {
+        let NetEvent::Gossip { data, .. } = evt;
+        if let Ok(DoubleEchoRequest {
+            request: Some(double_echo_request),
+        }) = DoubleEchoRequest::decode(&data[..])
+        {
+            match double_echo_request {
+                double_echo_request::Request::Gossip(Gossip {
+                    certificate: Some(certificate),
+                }) => match uci::Certificate::try_from(certificate) {
+                    Ok(cert) => {
                         let channel = self.tce_cli.get_double_echo_channel();
+                        if let hash_map::Entry::Vacant(entry) = self.delivery_latency.entry(cert.id)
+                        {
+                            entry.insert(CERTIFICATE_DELIVERY_LATENCY.start_timer());
+                        }
+
                         spawn(async move {
-                            let certificate_id = certificate_id.clone().try_into().map_err(|e| {
-                                error!(
-                                    "Invalid certificate id, could not process Echo message: {e}, \
-                                     certificate_id: {certificate_id}"
-                                );
-                                e
-                            });
-                            let validator_id = validator_id.clone().try_into().map_err(|e| {
+                            info!("Send certificate {} to be broadcast", cert.id);
```

Comment:
Is `INFO` the right level? Seems a bit verbose to me?

Reply:
Proved pretty useful so far. Debug is too verbose for short inspection if tce works because too many dependency crates opens their debug logs and you can not see the tree from the forest